Addons/tsgExoticQuadrature.hpp

```diff
@@ -45,30 +45,40 @@ inline double poly_eval(const std::vector<double> &roots, double x) {
     return eval_value;
 }
 
-//! \brief Generate the roots of all orthogonal polynomials up to degree n, where orthogonality is with respect to the measure
-//! given by ref_weights. It is assumed that the values of ref_weights are nonnegative.
+//! \brief Generate the roots of all orthogonal polynomials up to degree n, where orthogonality is with respect to the
+// measure given by ref_weights. Specializes depending on if the underlying weight function is symmetric or not.
+template <bool is_symmetric>
 inline std::vector<std::vector<double>> getRoots(const int n, const std::vector<double> &ref_weights,
                                                  const std::vector<double> &ref_points) {
 
     // Compute the roots incrementally.
     assert(ref_points.size() == ref_weights.size());
     assert(ref_points.size() > 0);
+    if (not std::all_of(ref_weights.begin(), ref_weights.end(), [](double x){return x>=0.0;})) {
```

Comment:
Right now, you are checking whether this is true in two places, here and in the external API call to `getExoticQuadrature()`. You only need one of those checks.

Reply:
I decided to just keep the test in ``getExoticQuadrature`` since ``getRoots`` should probably remain internal (and hence, its errors should not be shown to the user)